### PR TITLE
Assignment1

### PR DESCRIPTION
**Pull request by RinorSefa**
Open for comments/suggestion

### CODE REVIEW

File: test/src/edu/stanford/nlp/math/SloppyMathTest.java [hunk 1]
```diff
@@ -9,14 +9,20 @@
...
+    assertEquals(0.0, SloppyMath.round(-0.49));
+//  assertEquals(-1.0, SloppyMath.round(-0.5));  it fails
     assertEquals(10.0, SloppyMath.round(10));
     assertEquals(10.0, SloppyMath.round(10.32));
   }
 
   public void testRound2() {
+      assertEquals(0.0, SloppyMath.round(49, -2));
+      assertEquals(100.0, SloppyMath.round(50, -2));
+      assertEquals(100.0, SloppyMath.round(51, -2));
```
Comment (julwil) [inline test/src/edu/stanford/nlp/math/SloppyMathTest.java]: looks like too much indentation @RinorSefa

File: test/src/edu/stanford/nlp/math/SloppyMathTest.java [hunk 2]
```diff
@@ -9,14 +9,20 @@
   public void setUp() {
   }
 
-  public void testRound1() {
+    public void testRoundWithBoundaries() {
```
Comment (julwil) [inline test/src/edu/stanford/nlp/math/SloppyMathTest.java]: also indentation @RinorSefa

File: test/src/edu/stanford/nlp/patterns/CandidatePhraseTest.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
+package edu.stanford.nlp.patterns;
+
+import edu.stanford.nlp.stats.ClassicCounter;
+import edu.stanford.nlp.stats.Counter;
+import org.junit.Assert;
+import org.junit.Test;
+
```
Comment (julwil) [inline test/src/edu/stanford/nlp/patterns/CandidatePhraseTest.java:7]: remove new line @RinorSefa

**Comment by jvdassen** [comment]
The code looks nice to me except for one minor indentation issue [here](https://github.com/sw-testing-lab-group-corenlp/CoreNLP/blob/b59cd727656fc7b1aec5c1fb67bef0a7d4ae1dad/test/src/edu/stanford/nlp/math/SloppyMathTest.java#L12).

Otherwise the only negative thing I see is that sometimes commit messages are over 80chars, something that is often disregarded. For example here:
https://github.com/sw-testing-lab-group-corenlp/CoreNLP/pull/2/commits/3798327cec522302d261393ff2cb4e8d21b2ce28

You could use the title line with 80chars and the rest in the body.
```git
Boundary testing for method isAcronym

- Added xy
- Did this that
- Found bug in method xy!
```
I guess its very minor but something we could do for next time?

File: test/src/edu/stanford/nlp/patterns/CandidatePhraseTest.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        //Get branch
+        candidatePhrase = CandidatePhrase.createOrGet("run", "runs");
+        Assert.assertEquals(candidatePhrase.getPhrase(), String.valueOf("run"));
+        Assert.assertEquals(candidatePhrase.getPhraseLemma(), String.valueOf("runs"));
+    }
+
+    @Test
+    public void testStaticMethodCreateOrGetWithLemmaAndCounterBranchCoverage() {
+
+        //get branch
```
Comment (julwil) [inline test/src/edu/stanford/nlp/patterns/CandidatePhraseTest.java]: we could delete this comment @RinorSefa

File: test/src/edu/stanford/nlp/patterns/CandidatePhraseTest.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        List<String> stringList = new LinkedList();
+        stringList.add("sleep");
+        stringList.add("eat");
+
+        List<CandidatePhrase> candidateList = CandidatePhrase.convertStringPhrases(stringList);
+        Assert.assertEquals(candidateList.get(0).getPhrase(), "sleep");
+        Assert.assertEquals(candidateList.get(1).getPhrase(), "eat");
+
+        Assert.assertEquals(stringList.size(), candidateList.size());
+
```
Comment (julwil) [inline test/src/edu/stanford/nlp/patterns/CandidatePhraseTest.java]: remove nl @RinorSefa

File: test/src/edu/stanford/nlp/util/AcronymMatcherTest.java [hunk 6]
```diff
@@ -19,6 +20,14 @@ public void testBasic() {
     assertTrue(AcronymMatcher.isAcronym("ECC", "Election Complaints Commission".split("\\s+")));
   }
 
+  @Test
+  public void testBasicWithBoundaries(){
+    assertTrue(AcronymMatcher.isAcronym("B", "B".split("\\s+")));
+    assertFalse(AcronymMatcher.isAcronym("", "B".split("\\s+")));
+//  assertFalse(AcronymMatcher.isAcronym("", "A".split("\\s+")));
```
Comment (julwil) [inline test/src/edu/stanford/nlp/util/AcronymMatcherTest.java:27]: @RinorSefa  is this also a failing test?

**Review by julwil — APPROVED** [review]
Looks good! I only found some smaller cosmetics :)

File: test/src/edu/stanford/nlp/util/AcronymMatcherTest.java (hunk 6 above)
Comment (jvdassen) [reply test/src/edu/stanford/nlp/util/AcronymMatcherTest.java:27]: Yes I believe that's the other bug he found

**Review by jvdassen — APPROVED** [review]

**Pull request merged**